server/cluster/cluster.go

```diff
@@ -69,8 +82,62 @@ func (s *Server) List(ctx context.Context, q *cluster.ClusterQuery) (*appv1.Clus
 	if err != nil {
 		return nil, err
 	}
-	clusterList.Items = items
-	return clusterList, nil
+
+	cl := *clusterList
+	cl.Items = items
+
+	return &cl, nil
```

Comment:
Any reason for this change? I think initial code for this should work as well.. Maybe revert this?

Reply:
Hi @ishitasequeira,

Thanks for taking a look at this PR. 

The reason for this change is to avoid mutating the original object (due to the pointer/reference). Without this, tests will fail since we will end up updating the [mockClusterList returned by our mock DB](https://github.com/onematchfox/argo-cd/blob/3b5f9c31cdbd66d8b8a16ad80f7777f158300897/server/cluster/cluster_test.go#L515-L520). 

FWIW, you can see similar behaviour [here](https://github.com/argoproj/argo-cd/blob/064c8da942c56bf45745d7bc1b9af8ded979aa61/server/application/application.go#L240-L246) on the application `List` endpoint. I'm happy to change to instantiating a new `ClusterList` (as is done in the linked example) if you prefer that approach. My reasoning for simply dereferencing the existing object was that it ensures that we have a full copy of the existing object. IMO, this simplifies maintenance going forward, as we don't need to be concerned about new fields being added to `ClusterList` that would then need to be added to this code as well if we were to create a new instance instead).

Reply:
@onematchfox Thanks for the detailed reasoning. With your explanation, the test does make sense. Thanks.